android/app/src/main/java/com/couchbase/todo/db/DAO.java

```diff
@@ -282,6 +290,28 @@ public void deleteById(@NonNull String docId) {
         catch (CouchbaseLiteException e) { reportError(e); }
     }
 
+    public List getAllDocIds() throws CouchbaseLiteException {
+        List<String> idList = new ArrayList<>();
+        Query query = createQuery(SelectResult.expression(Meta.id));
+        ResultSet idsSet = query.execute();
+        for (Result id = idsSet.next(); id != null; id = idsSet.next()) { idList.add(id.getString(0)); }
+        return idList;
+    }
+
+    /*
+    get ids of all tasks from a list
+    return list of id
+     */
+    @WorkerThread
+    public List getTaskIdsFromList(String listID) throws CouchbaseLiteException {
+        List<String> taskIds = new ArrayList<>();
+        Query query = createQuery(SelectResult.expression(Meta.id))
+            .where(Expression.property("type").equalTo(Expression.string("task"))
+                .and(Expression.property("taskList.id").equalTo(Expression.string(listID))));
+        ResultSet allTaskIds = query.execute();
```

Comment:
Just a thought.... you do this in both methods.  Perhaps you should pull it out into a utility method?

Reply:
By utility method you mean I should create a utility class and have a static method, or just a helper method for the repeated code? I think having a private helper method would be enough right?